### ISSUE TITLE
Docker image can't be built

### ISSUE THREAD

**Issue by wojciechkumon** [sanjP10/release-action#7]
### Prerequisites

* [x] Are you running the latest version?
* [x] Did you read through the instructions on the README?

# Describe the bug
Building `Dockerfile` doesn't work anymore. I found it using my release pipeline but I also reproduced it via simply running `docker build .` in the repository.

# To Reproduce
Run `docker build .`

output:
```
 % docker build .
[+] Building 4.6s (6/7)                                                                                                                                                     
 => [internal] load build definition from Dockerfile                                                                                                                   0.0s
 => => transferring dockerfile: 37B                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/golang:1.15.8-alpine                                                                                                0.2s
 => [internal] load build context                                                                                                                                      0.0s
 => => transferring context: 35B                                                                                                                                       0.0s
 => CACHED [1/3] FROM docker.io/library/golang:1.15.8-alpine@sha256:e3685a09bcee71a4d2be053f082687098da60ee21d36bb42828958cccea7df79                                   0.0s
 => ERROR [2/3] RUN GO111MODULE=on go get -u github.com/sanjP10/release@3.2.0                                                                                          4.4s
------                                                                                                                                                                      
 > [2/3] RUN GO111MODULE=on go get -u github.com/sanjP10/release@3.2.0:                                                                                                     
#5 0.238 go: downloading github.com/sanjP10/release v0.0.0-20220106083849-9513d660e4d6                                                                                      
#5 0.431 go: github.com/sanjP10/release 3.2.0 => v0.0.0-20220106083849-9513d660e4d6                                                                                         
#5 0.937 go: downloading github.com/google/subcommands v1.2.0                                                                                                               
#5 0.940 go: downloading github.com/go-git/go-git/v5 v5.4.2                                                                                                                 
#5 0.992 go: downloading github.com/hashicorp/go-version v1.4.0
#5 1.068 go: downloading github.com/mitchellh/go-homedir v1.1.0
#5 1.069 go: downloading github.com/go-git/gcfg v1.5.0
#5 1.069 go: downloading github.com/go-git/go-billy/v5 v5.3.1
#5 1.069 go: downloading golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b
#5 1.070 go: downloading github.com/emirpasic/gods v1.12.0
#5 1.070 go: downloading github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7
#5 1.070 go: downloading github.com/xanzy/ssh-agent v0.3.0
#5 1.071 go: downloading github.com/acomagu/bufpipe v1.0.3
#5 1.071 go: downloading github.com/imdario/mergo v0.3.12
#5 1.091 go: downloading github.com/sergi/go-diff v1.1.0
#5 1.092 go: downloading github.com/jbenet/go-context v0.0.0-20150711004518-d14ea06fba99
#5 1.096 go: downloading gopkg.in/warnings.v0 v0.1.2
#5 1.103 go: downloading github.com/Microsoft/go-winio v0.4.16
#5 1.111 go: downloading golang.org/x/sys v0.0.0-20210502180810-71e4cd670f79
#5 1.138 go: downloading golang.org/x/net v0.0.0-20210326060303-6b1517762897
#5 1.285 go: downloading github.com/kevinburke/ssh_config v0.0.0-20201106050909-4977a11b4351
#5 1.796 go: golang.org/x/sys upgrade => v0.0.0-20220330033206-e17cdc41300f
#5 1.796 go: golang.org/x/net upgrade => v0.0.0-20220325170049-de3da57026de
#5 1.875 go: github.com/xanzy/ssh-agent upgrade => v0.3.1
#5 1.892 go: github.com/sergi/go-diff upgrade => v1.2.0
#5 1.893 go: github.com/ProtonMail/go-crypto upgrade => v0.0.0-20220113124808-70ae35bab23f
#5 1.940 go: github.com/Microsoft/go-winio upgrade => v0.5.2
#5 1.969 go: github.com/go-git/go-billy/v5 upgrade => v5.3.1
#5 1.993 go: golang.org/x/crypto upgrade => v0.0.0-20220331220935-ae2d96664a29
#5 2.078 go: github.com/kevinburke/ssh_config upgrade => v1.2.0
#5 2.162 go: downloading github.com/sergi/go-diff v1.2.0
#5 2.163 go: downloading golang.org/x/sys v0.0.0-20220330033206-e17cdc41300f
#5 2.163 go: downloading github.com/ProtonMail/go-crypto v0.0.0-20220113124808-70ae35bab23f
#5 2.166 go: downloading github.com/xanzy/ssh-agent v0.3.1
#5 2.167 go: downloading golang.org/x/net v0.0.0-20220325170049-de3da57026de
#5 2.169 go: downloading golang.org/x/crypto v0.0.0-20220331220935-ae2d96664a29
#5 2.185 go: downloading github.com/kevinburke/ssh_config v1.2.0
#5 2.406 go: downloading github.com/Microsoft/go-winio v0.5.2
#5 3.707 # github.com/kevinburke/ssh_config
#5 3.707 pkg/mod/github.com/kevinburke/ssh_config@v1.2.0/config.go:281:12: undefined: os.ReadFile
#5 3.707 pkg/mod/github.com/kevinburke/ssh_config@v1.2.0/config.go:296:12: undefined: io.ReadAll
------
executor failed running [/bin/sh -c GO111MODULE=on go get -u github.com/sanjP10/release@3.2.0]: exit code: 2
```

# Expected behaviour
Docker image successfully built

# Initial investigation

https://github.com/kevinburke/ssh_config released version 1.2.0 today which doesn't work. `ssh_config` uses `GO111MODULE=off` in their pipeline and it's working, `release-action` uses `GO111MODULE=on`.
Suspected commit where it stopped working with `release-action`: https://github.com/kevinburke/ssh_config/commit/6ad71ac26ef41b372c2f0ca7839ec35e8f1a4235
I also tried to switch `GO111MODULE` to `off` in `release-action` `Dockerfile` locally but I got another errors (I'm not familiar with go environment):
```
 => ERROR [2/3] RUN GO111MODULE=off go get -u github.com/sanjP10/release@3.2.0                                                                                         0.2s
------
 > [2/3] RUN GO111MODULE=off go get -u github.com/sanjP10/release@3.2.0:
#5 0.233 go: cannot use path@version syntax in GOPATH mode
------
executor failed running [/bin/sh -c GO111MODULE=off go get -u github.com/sanjP10/release@3.2.0]: exit code: 1
```

**Comment by sanjP10** [comment]
@wojciechkumon Im gonna work on this at the weekend, apologies this is a package used by go-git which I consume. Will work on this ASAP!

**Comment by sanjP10** [comment]
@wojciechkumon Fixed, ssh_config an underlying module used from go-git seemed to be the issue. I updated the release [sanjP10/release ](https://github.com/sanjP10/release) tool to use go 1.18 and this seemed to work fine. So I bumped the version used and the container version used. 

The action still uses 1.0.0, as it was completely broken due to this issue, and I saw no need to ask everyone to update their pipelines.
